test/test_runner_battle.c

```diff
@@ -1332,6 +1332,10 @@ static void TearDownBattle(void)
     FreeBattleSpritesData();
     FreeBattleResources();
     FreeAllWindowBuffers();
+    if (DATA.setFlag != 0) {
+        FlagClear(DATA.setFlag);
+        DATA.setFlag = 0;
+    }
```

Comment:
Oh! I think `TearDownBattle` doesn't always run? It seems it's only done when a test fails unexpectedly (because that's what sets `STATE->tearDownBattle = TRUE`), or between `PASSES_RANDOMLY` trials?

This logic probably wants to be in its own function which is called from `BattleTest_TearDown`, `CB2_BattleTest_NextParameter`, and `CB2_BattleTest_NextTrial`?